python/py-pip/Portfile

```diff
@@ -12,7 +12,7 @@ platforms           darwin
 license             MIT
 supported_archs     noarch
 
-python.versions     26 27 33 34 35 36 37
+python.versions     pypy pypy3 26 27 34 35 36 37
```

Comment:
I would prefer calling the 2.7-compatible pypy something like `pypy2` to convey the language version. On a related point, having to rev bump on a per-subport basis when a pypy's language version changes seems really inconvenient, but I guess it would be liveable if it's only a small number of ports. (But OTOH if only a small number of ports need these changes, should the portgroup be involved at all?)

Reply:
> I would prefer calling the 2.7-compatible pypy something like pypy2 to convey the language version.

That'd be slightly inconsistent with the naming of the ports, but I guess it makes sense. I considered renaming the `pypy` port to `pypy2`, but left it as-is to avoid unnecessary breakage.

> On a related point, having to rev bump on a per-subport basis when a pypy's language version changes seems really inconvenient, but I guess it would be liveable if it's only a small number of ports. (But OTOH if only a small number of ports need these changes, should the portgroup be involved at all?)

Well, there's no particular reason that the number of ports with PyPy support should be small. For example, I'd like to be able to install IPython for PyPy, and that one depends on 27 other Python ports…

My general idea was that the PortGroup itself would bump the `epoch` if needed. I know it's vaguely horrible, but hopefully we won't need to do it too often. Essentially, all non-`noarch` ports would (probably) need it on every major PyPy upgrade, and all ports might need it for a Python language upgrade in pypy3.